packages/core/src/test/amazonqFeatureDev/controllers/chat/controller.test.ts

```diff
@@ -323,4 +329,87 @@ describe('Controller', () => {
             assert.strictEqual(revertRejection.state.filePaths?.find(i => i.relativePath === filePath)?.rejected, false)
         })
     })
+
+    describe('insertCode', () => {
+        it('sets the number of files accepted counting also deleted files', async () => {
+            async function insertCode() {
+                const initialState = new PrepareCodeGenState(
+                    {
+                        conversationId: conversationID,
+                        proxyClient: new FeatureDevClient(),
+                        workspaceRoots: [''],
+                        workspaceFolders: [controllerSetup.workspaceFolder],
+                        uploadId: uploadID,
+                    },
+                    '',
+                    [
+                        {
+                            zipFilePath: 'myfile1.js',
+                            relativePath: 'myfile1.js',
+                            fileContent: '',
+                            rejected: false,
+                            virtualMemoryUri: '' as unknown as vscode.Uri,
+                            workspaceFolder: controllerSetup.workspaceFolder,
+                        },
+                        {
+                            zipFilePath: 'myfile2.js',
+                            relativePath: 'myfile2.js',
+                            fileContent: '',
+                            rejected: true,
+                            virtualMemoryUri: '' as unknown as vscode.Uri,
```

Comment:
I don't recommend setting vritualMemoryUri like this.

you can use   generateVirtualMemoryUri() function like in this test [link](https://github.com/aws/aws-toolkit-vscode/pull/4594/files)

Note: I doesn't effect the test functionality, but still better to keep code as close to "actual" implementation as possible, but cool that you added a test!